### PR TITLE
Add test name formatter fn

### PR DESCRIPTION
**Pull request by dem4ron**
Hi!

I'm opening this PR to start a conversation about finding a good way to format the test name outputs.

There is currently this issue that I'd like to resolve: https://github.com/exercism/exercism/issues/6544

I wrote a small fn to solve this. I'm not a Rust dev (hopefully one day!) but wanted to propose a solution. Please review my code, and give suggestions on how to improve it. 

Thank you

### CODE REVIEW

**Review by ErikSchierboom — COMMENTED** [review]
This will be a great improvement. The main thing that is lacking are (and you are already aware of that) tests. Could you add those? The way you would do that is to add a new directory/test case to the [`tests` directory](https://github.com/exercism/rust-test-runner/tree/main/tests) and make sure that there many different types of test names being tested.

File: tests/test-name-formatting/expected_results.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
+  "version": 2,
+  "status": "pass",
+  "message": null,
+  "tests": [
+    {
+      "name": "test_dont_replace_hidden_test",
```
Comment (ErikSchierboom) [inline tests/test-name-formatting/expected_results.json:7]: It looks like these names aren't actually transformed?

**Review by ErikSchierboom — APPROVED** [review]
LGTM! 

@petertseng Could you look at the rust code?

File: tests/test-name-formatting/tests/formatter.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#[test]
+fn test_extra_underscore_middle() {
+    let input =
+        String::from("test_valid_strings_with____a_nondigit_added_at_the_end_become_invalid");
+    let output = String::from("Valid strings with a nondigit added at the end become invalid");
+    assert_eq!(formatter::format_test_name(input), output);
+}
+
+#[test]
+fn test_several_extra_whitespaces() {
```
Comment (dem4ron) [inline tests/test-name-formatting/tests/formatter.rs:45]: (ik, this is an unrealistic scenario)

File: tests/test-name-formatting/tests/formatter.rs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+#[test]
+fn test_sponsored_by() {
+    let input = String::from("_ _ _apple_ _ _");
+    let output = String::from("Apple");
+    assert_eq!(formatter::format_test_name(input), output);
+}
+
+#[test]
+fn test_only_remove_first_test() {
```
Comment (dem4ron) [inline tests/test-name-formatting/tests/formatter.rs:82]: this one tested a different implementation of `"test"` removal

File: tests/test-name-formatting/tests/formatter.rs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    let input = String::from("test_only_remove_first_test");
+    let output = String::from("Only remove first test");
+    assert_eq!(formatter::format_test_name(input), output);
+}
+
+#[test]
+fn test_dont_replace_hidden_test() {
```
Comment (dem4ron) [inline tests/test-name-formatting/tests/formatter.rs:89]: so did this one.

File: tests/test-name-formatting/tests/formatter.rs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    let input = String::from("protest_against_bad_tests");
```
Comment (petertseng) [inline tests/test-name-formatting/tests/formatter.rs:90]: oh! I didn't know they felt so strongly that they need to hold a protest!

File: tests/test-name-formatting/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+pub fn format_test_name(name: String) -> String {
+    let name = name.to_lowercase().replace("_", " ");
+    let mut name:Vec<_> = name.split_whitespace().collect();
+    if name[0] == "test"{
```
Comment (petertseng) [inline tests/test-name-formatting/src/lib.rs:4]:
```suggestion
    if name[0] == "test" {
```

I see that the formatter is not automatically run on files in this repository.

**Review by petertseng — COMMENTED** [review]
I have reviewed the Rust code and everything is good as it is.
I have also reviewed some non-Rust portions, which I left some comments on, but none of them are considered required to be addressed before this is merged

**Comment by dem4ron** [comment]
> I have reviewed the Rust code and everything is good as it is. I have also reviewed some non-Rust portions, which I left some comments on, but none of them are considered required to be addressed before this is merged

@petertseng Thank you for your review, I'll make changes to the tests, and check the formatter.

**Comment by ErikSchierboom** [comment]
> I have reviewed the Rust code and everything is good as it is.
> I have also reviewed some non-Rust portions, which I left some comments on, but none of them are considered required to be addressed before this is merged

Thanks Peter!

**Pull request merged**